packages/grid/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -245,12 +245,12 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>((props, ref) =>
   const isSectionLastCell = sectionIndex === sectionLength - 1;
 
   const showLeftBorder =
```

Comment:
@romgrk I think this change results in a double vertical border when `showCellVerticalBorder={true}`:

https://app.argos-ci.com/mui/mui-x/builds/16834/70974455
https://app.argos-ci.com/mui/mui-x/builds/16834/70974454